pkg/cli/image_apk.go

```diff
@@ -0,0 +1,245 @@
+	cmd := &cobra.Command{
+		Use:   "apk <image>",
+		Short: "Show APK(s) in a container image",
+		Example: `
+  # Show all APKs in an image
+  wolfictl image apk cgr.dev/chainguard/bash
+
+  # Show all APKs in an image that own a component (based on a Syft analysis)
+  wolfictl image apk cgr.dev/chainguard/coredns -c 'github.com/aws/aws-sdk-go'
+
+  # Show all APKs in an image that own a component, and show the path to the
+  # Melange configuration file for each APK, within the given directory
+  wolfictl image apk cgr.dev/chainguard/prometheus-operator -c 'github.com/aws/aws-sdk-go' -d ~/code/wolfi-os
+`,
+		SilenceErrors: true,
+		Args:          cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			imageRef := args[0]
+
+			imgSource, err := source.NewFromStereoscopeImage(source.StereoscopeImageConfig{
+				Reference: imageRef,
+				From:      image.OciRegistrySource,
+			})
+			if err != nil {
+				return fmt.Errorf("unable to construct scan source for image %q: %w", imageRef, err)
+			}
+			defer imgSource.Close()
+
+			cfg := syft.DefaultCreateSBOMConfig()
+			imgSBOM, err := syft.CreateSBOM(cmd.Context(), imgSource, cfg)
+			if err != nil {
+				return fmt.Errorf("unable to create SBOM: %w", err)
+			}
+
+			distroID := imgSBOM.Artifacts.LinuxDistribution.ID
+			if !slices.Contains([]string{"wolfi", "chainguard"}, distroID) {
+				return fmt.Errorf("unsupported distro: %s", distroID)
+			}
+
+			r := &syftResults{sbom: imgSBOM}
+
+			var apks []pkg.Package
+
+			if c := p.component; c != "" {
+				apks, err = r.apksOwningPackageWithName(c)
+				if err != nil {
+					return fmt.Errorf("unable to find APK package owning a component %q: %w", c, err)
+				}
+			} else {
+				apks = r.apks()
+			}
+
+			if len(p.distroDirPaths) == 0 {
+				for i := range apks {
+					apk := apks[i]
+					fmt.Println(apk.Name)
+				}
+
+				return nil
+			}
+
+			configPaths, err := apkConfigPaths(apks, p.distroDirPaths)
+			if err != nil {
+				return fmt.Errorf("unable to find configuration files for APK packages: %w", err)
+			}
+
+			for _, p := range configPaths {
+				fmt.Println(p)
+			}
+
+			return nil
+		},
+	}
+
+	p.addFlags(cmd)
+	return cmd
+}
+
+type imageAPKParams struct {
+	component      string
+	distroDirPaths []string
+}
+
+func (p *imageAPKParams) addFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&p.component, "component", "c", "", "show only APKs containing the given component")
+	cmd.Flags().StringSliceVarP(&p.distroDirPaths, "distro-dir", "d", nil, "path to a directory containing Melange build configuration files")
+}
+
+type syftResults struct {
+	sbom                                *sbom.SBOM
+	packageOwnershipsByOwnedPackageName map[string][]ownership
+}
+
+type ownership struct {
+	owner pkg.Package
+	owned pkg.Package
+}
+
+func (r *syftResults) apks() []pkg.Package {
+	var apks []pkg.Package
+
+	pkgs := r.sbom.Artifacts.Packages.Sorted(pkg.ApkPkg)
+	for i := range pkgs {
+		p := pkgs[i]
```

Comment:
Is this better? More of a style question, but I wonder if it has different implications on the code below

```suggestion
	for i, p := range pkgs {
```

Reply:
Switched back to that and added a `nolint` — also happy to strike the lint rule globally if we want!